test-vectors/src/peg_handoff.rs

```diff
@@ -0,0 +1,49 @@
+use bitcoin::{
+    opcodes, script::Builder, OutPoint, Script, Sequence, Transaction, TxIn, TxOut, Witness,
+};
+
+use crate::utils::generate_test_vector;
+
+pub fn generate_peg_handoff_test_vector() -> Transaction {
+    let input = TxIn {
+        previous_output: OutPoint::null(),
+        script_sig: Script::empty().into(),
+        sequence: Sequence(65535),
+        witness: Witness::new(),
+    };
+
+    let input = vec![input];
+
+    let peg_wallet_address = [4; 32];
+    let reward_cycle = 67;
+
+    let output = generate_peg_handoff_output(peg_wallet_address, reward_cycle);
+
+    generate_test_vector(input, output)
+}
+
+fn generate_peg_handoff_output(peg_wallet_address: [u8; 32], reward_cycle: u8) -> Vec<TxOut> {
+    let op_bytes = [105, 100, b'H'];
+
+    let op_return_script = Builder::new()
+        .push_opcode(opcodes::all::OP_RETURN)
+        .push_slice::<[u8; 3]>(op_bytes)
+        .push_slice::<[u8; 1]>([reward_cycle])
```

Comment:
Nice with the explicit slice size annotations :star:

Reply:
The reward cycle should be a u64 though.
```suggestion
        .push_slice::<[u8; 3]>(op_bytes)
        .push_slice(reward_cycle.as_be_bytes())
```

Reply:
The reward cycle should be a u64 though.
```suggestion
        .push_slice::<[u8; 3]>(op_bytes)
        .push_slice(reward_cycle.as_be_bytes())
```